ci-operator/step-registry/rosa/nodes/wait-ready/rosa-nodes-wait-ready-commands.sh

```diff
@@ -10,24 +10,106 @@ log(){
+    log "$(date) - Worker nodes not in Ready state, if any"
+    for node in $(oc get nodes --no-headers -l node-role.kubernetes.io/worker,node-role.kubernetes.io/infra!= --output jsonpath="{.items[?(@.status.conditions[-1].type!='Ready')].metadata.name}"); do
+      oc describe node "$node"
+    done
+    log "$(date) - Finished printing details of all infra and worker nodes."
+}
+
+# Display details of machinesets and nodes for Classic Rosa cluster
+function listMachineAndNodeDetails() {
+    echo "List machine details"
+    # Get current machinesets and machines 
     log "$(date) - List of machinesets"
-    oc get machinesets -A
+    echo "oc get machinesets -A -n openshift-machine-api"
+    oc get machinesets -A -n openshift-machine-api
     log "$(date) - List of machines"
-    oc get machines -A | cat -n
-    log "$(date) - List of all worker only nodes, excluding infra"
-    oc get nodes --no-headers -l node-role.kubernetes.io/worker,node-role.kubernetes.io/infra!= | cat -n
+    echo "oc get machines -A -n openshift-machine-api | cat -n"
+    oc get machines -A -n openshift-machine-api | cat -n
 
-    # Get details of machines not in Running state
+    # Get details of worker machines not in Running state
+    log "$(date) - Worker machines not in Running state, if any"
     for machine in $(oc get machine  -n openshift-machine-api -l machine.openshift.io/cluster-api-machine-type=worker,machine.openshift.io/cluster-api-machine-type!=infra  --output jsonpath="{.items[?(@.status.phase!='Running')].metadata.name}"); do
       oc describe machine "$machine" -n openshift-machine-api
     done
-    # Get details of nodes not in Ready state
-    for node in $(oc get nodes --no-headers -l node-role.kubernetes.io/worker,node-role.kubernetes.io/infra!= --output jsonpath="{.items[?(@.status.conditions[-1].type!='Ready')].metadata.name}"); do
-      oc describe node "$node"
+    echo
+    
+    # Get details of all nodes 
+    listNodeDetails
+}
+
+# Display Machine Config Pool details
+function listMachineConfigPoolDetails() {
+    log "$(date) - List machine config pools"
+    echo "oc get mcp"
+    oc get mcp
+}
+
+# List details of machinesets, machines and nodes depending on Classic Rosa or HCP cluster
+function listDetails() {
+  listMachineConfigPoolDetails
+  if [ "$is_hcp_cluster" = "false" ]; then
+    echo "Listing machine and node details"
+    listMachineAndNodeDetails
+  else
+    echo "Listing node details only"
+    listNodeDetails
+  fi
+}
+
+# Function introducing sleep and wait for worker nodes to become Ready state
+function waitForReady() {
+    echo "Wait for all nodes to be Ready"
+
+    # Sleep for 2 minutes to allow for the nodes to begin restarting after initial scaling
+    sleep 120
+    # Query the node state until all of the nodes are ready
+    for i in {1..60}; do
+        NODE_STATE="$(oc get nodes || echo "ERROR")"
+        if [[ ${NODE_STATE} == *"NotReady"*  || ${NODE_STATE} == *"SchedulingDisabled"* ]]; then
```

Comment:
I'd like to judge the not-ready node number is NOT 0 via `oc get nodes --no-headers -l node-role.kubernetes.io/worker,node-role.kubernetes.io/infra!= --output jsonpath="{.items[?(@.status.conditions[-1].type!='Ready') | wc -w | xargs`

Reply:
I think that a node can have the Ready status condition and still be SchedulingDisabled. The only way to see if a node is scheduling disabled is to look at the taint on a node 

`Taints:             node.kubernetes.io/unschedulable:NoSchedule`

Reply:
For now, I noticed that current code is working to check for both Ready and SchedulingDisabled.

Reply:
I meant based on the sugguested change it wont check for the scheduling disabled like your code currently covers/checks

Reply:
Ah, gotcha! Haven't completely read Yanming's suggestions yet :)

Reply:
@yasun1 One of the things we found out during testing was, sometimes nodes are marked Ready, but they are still marked as "SchedulingDisabled". So, what we really want to check after scaling and before starting workload is, a) all nodes are Ready and b) They are not in SchedulingDisabled state.
Hence, the check here incorporates both.

Reply:
In this step, we are checking Node states and the actual count verification is done in a different step.